lib/remote_persistent_term/fetcher/http/cache.ex

```diff
@@ -0,0 +1,54 @@
+defmodule RemotePersistentTerm.Fetcher.Http.Cache do
+  @moduledoc """
+  A module to assist in following the HTTP Caching spec.
+
+  See more: https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching
+  """
+
+  @max_age_regex ~r/max-age=(\d+)/
+  @default_age 0
+
+  @doc """
+  Calculates the refresh interval from the cache-control and age headers.
+  Caching directives are case-insensitive. Multiple directives are permitted and must be comma-separated
+
+  Refresh interval is in milliseconds
+  """
+  @spec refresh_interval(Req.Response.t()) :: {:ok, pos_integer()} | {:error, term()}
+  def refresh_interval(resp) do
+    with {:ok, max_age} <- max_age(resp) do
+      age = age(resp)
+      refresh_interval = (max_age - age) |> validate_refresh_interval() |> :timer.seconds()
+      {:ok, refresh_interval}
+    end
+  end
+
+  defp max_age(resp) do
+    with [cache_control | _] <- Req.Response.get_header(resp, "cache-control"),
+         true <- is_binary(cache_control) do
```

Comment:
```suggestion
    with [cache_control | _] <- Req.Response.get_header(resp, "cache-control") do
```
https://hexdocs.pm/req/Req.Response.html#get_header/2 only returns a list of binaries, so we don't need to check that it is a binaries, just that it is not empty